foursight_core/run_result.py

```diff
@@ -158,6 +158,8 @@ def record_run_info(self):
         Returns True on success, False otherwise
         """
         run_id = self.kwargs['_run_info']['run_id']
+        if not hasattr(self, 'prefix'):
+            raise Exception("foursight prefix must be defined using set_prefix")
```

Comment:
Since we are in core, I would consider defining exception classes and using them in situations like this (rather than just throwing Exception).